File: Java/src/1773_CountItemsMatchingARule/CountItemsMatchingARule.java
import java.util.List;

class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int ans = 0 ;
        if(ruleKey.equals("type")){
            for(List<String>  stringList : items){
                if(stringList.get(0).equals(ruleValue))
                    ans++;
            }
        }else if(ruleKey.equals("color")){
            for(List<String>  stringList : items){
                if(stringList.get(1).equals(ruleValue))
                    ans++;
            }
        }else {
            for(List<String>  stringList : items){
                if(stringList.get(2).equals(ruleValue))
                    ans++;
            }
        }
        return ans;
    }
}

File: golang/1773_CountItemsMatchingARule/countItemsMatchingARule.go
package countitemsmatchingarule

func countMatches(items [][]string, ruleKey string, ruleValue string) int {
	ans := 0
	rule := map[string]int{"type": 0, "color": 1, "name": 2}[ruleKey]
	for _, item := range items {
		if item[rule] == ruleValue {
			ans++
		}
	}
	return ans
}
